### DISCUSSION TITLE
how to config my own lua_shared_dict ?

### DISCUSSION THREAD

**Discussion by Git4Future** [apache/apisix#6150] [Q&A] [answered]
**i have already add my config in config.yml ,but when i reload apisix, i can't find it in nginx.conf** 
 
**config-default.yml**
lua_shared_dict:
       ext-plugin: 1m
       mydict: 10m

**Comment by starsz** [comment]
Hi @Git4Future Where do you add your config?

You should put it under `custom_lua_shared_dict`

Refer: https://github.com/apache/apisix/blob/66f66d499ab7facf51a3251e9c1739c8b7fabd80/conf/config-default.yaml#L231-L232

**Reply by Git4Future** [reply]
thank you

**Comment by leslie-tsang** [comment, marked as answer]
Hello there, please use `custom_lua_shared_dict` instead of `lua_shared_dict`, and make sure you indent it correctly.
It's hard to figure out what's going on in your case because of the description.
Normally, you should define your own `shared dict` in [config.yaml](https://github.com/apache/apisix/blob/master/conf/config.yaml), as follows
```yaml
...
nginx_config:
    ...
    custom_lua_shared_dict:
        mydict: 10m
...
```

**Reply by Git4Future** [reply]
Thanks a lot, i put my own config under  `lua_shared_dict` by mistake !
the description of **config-default.yml** is confused ，It took me a lot of time to figure out,until i find the key propertie in **/apisix/apisix/cli/ngx_tpl.lua** and **/apisix/apisix/cli/ops.lua**

**Reply by leslie-tsang** [reply]
Glad to see the problem is solved, 🥳🥳
> the description of config-default.yml is confused ，It took me a lot of time to figure out,until i find the key propertie in /apisix/apisix/cli/ngx_tpl.lua and /apisix/apisix/cli/ops.lua

`config-default.yaml` is more like a fallback configuration template, APISIX cli tool will merge `config.yaml` and `config-default.yaml` as the **final read-in configuration**, and what is defined in `config.yaml` will take effect first.
